### PR TITLE
Add Count() function

### PR DESCRIPTION
**Pull request by lzap**
Hey,

I was browsing the code, looks nice. However, isn't sorting for deletion slow? Iteration through just once is faster.

The Keys() function exposing sorted elements is dangerous, I almost used it to do counting of keys. Therefore I added a Count function to do the same, for monitoring/telemetry purposes.

Is cleaning code finished? It does not appear to be, `janitor.run()` is not called from the cache at all, only from test. I guess this is still WIP.

Cheers!

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.idea/
```
Comment (lzap) [inline .idea/.gitignore:1]: Damn, I would drop this if you like the patch.

File: cache.go [hunk 2]
```diff
@@ -219,13 +229,18 @@ func (c *Cache[K, V]) Set(key K, val V, opts ...ItemOption) {
...
+// Keys returns the keys of the cache. The order is relied on algorithms.
 func (c *Cache[K, V]) Keys() []K {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	return c.cache.Keys()
 }
 
+// Size returns number of items in the cache.
+func (c *Cache[K, V]) Size() int {
+	return c.cache.Size()
```
Comment (Code-Hex) [inline cache.go:241]: required mutex
```suggestion
	c.mu.Lock()
	defer c.mu.Unlock()
	return c.cache.Size()
```

File: policy/clock/clock.go [hunk 3]
```diff
@@ -138,3 +138,8 @@ func (c *Cache[K, V]) Delete(key K) {
 func (c *Cache[K, V]) Len() int {
 	return len(c.items)
 }
+
+// Size returns number of items in the cache.
+func (c *Cache[K, V]) Size() int {
+	return len(c.items)
+}
```
Comment (Code-Hex) [inline policy/clock/clock.go:145]: I think you can use each Len method instead of re-adding Size methods

**Comment by Code-Hex** [comment]
@lzap Thanks for your PR!

Ah... I need to call run method. Thanks

> Is cleaning code finished? It does not appear to be, janitor.run() is not called from the cache at all, only from test. I guess this is still WIP.

**Comment by lzap** [comment]
Okay, I ended up writing my own implementation, I have some specific needs (Prometheus monitoring). I see conflicts, closing! Good luck with the project.

**Pull request closed without merging**